### PR TITLE
Fixes #137: Prevent non-required fields to fill state with defaults.

### LINKED ISSUE #137: non required number type field starts off with a value of 0
### Description

When a field has a type number and is not required, the starting value for the field is 0. Since 0 is possibly a legitimate value, shouldn't this field be empty by default instead?

Here is the output for JSON.stringify(props):
```
{"name":"temperature",
"required":false,
"schema":{"description":"Temperature of the asset in CELSIUS.","type":"number"},
"idSchema":{"id":"root_temperature"},
"formData":0,
"registry":{"fields":{},"widgets":{},"definitions":{}}}
```

#### Expected behavior

The field should start off as an empty field.

#### Actual behavior

The field starts off with a value of 0.

### Version

"react-jsonschema-form": "^0.21.0"

### PR DESCRIPTION
**Pull request by n1k0**
Refs #137, /cc @alexmnguyen.

We should carefully review this one at is has possible strong impact on forms behavior, though couldn't spot any problem in tests and in the playground.

r=? @leplatrem @magopian

### CODE REVIEW

**Comment by n1k0** [comment]
I'm gonna move the default definition for datetime into the widget itself

**Comment by n1k0** [comment]
Deployed the changes to gh-pages http://mozilla-services.github.io/react-jsonschema-form/

File: src/utils.js [hunk 1]
```diff
@@ -66,28 +66,8 @@ export function getDefaultRegistry() {
...
-  }
-  const {type} = schema;
-  switch (type) {
-  case "array":     return [];
-  case "boolean":   return false;
-  case "number":    return 0;
-  case "object":    return {};
-  case "string":    {
-    if (schema.format === "date-time") {
-      return new Date().toJSON();
```
Comment (leplatrem) [inline src/utils.js]: That particular one is a breaking change, no ?

How do we get date-time fields to default to current date ? It should take the current datetime, not the one when the app/schema was instantiated.

**Comment by leplatrem** [comment]
Sorry I hadn't seen your comment about `parseDateString`.

**Comment by leplatrem** [comment]
it's a pity that we're obliged to repeat the ``sandbox.stub(console, "error", (error)`` in every test suite.

r+ otherwise :)

(as long as the default datetime is still «now»)

**Comment by n1k0** [comment]
> it's a pity that we're obliged to repeat the sandbox.stub(console, "error", (error) in every test suite.

Factored this.

I'm gonna land the patch after #132 is landed, as it probably conflicts heavily.

**Comment by n1k0** [comment]
I need to land this now and make a release as this fixes a bunch of issues we need on other projects.

**Pull request merged**